parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,336 @@
+        }
+        if (!statement.trim().equals(unparsed.trim())) {
+            return Optional.of(UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR);
+        }
+        return Optional.empty();
+    }
+
+    private static void writeTestResultToFile(SqlLogicTestResult result, Path resultPath, String nameOfResultFile)
+    {
+        try (PrintWriter writer = new PrintWriter(resultPath.resolve(nameOfResultFile).toString())) {
+            for (SqlLogicTestStatement statement : result.getErroneousStatements()) {
+                writer.print(statement.getStatement() + " ----> " + statement.getError().get().name());
+                writer.println();
+                writer.println();
+            }
+            writer.println("===============================================");
+            writer.println("RESULT");
+            writer.println("===============================================");
+            writer.printf("Database dialect tested against: %s", result.getDatabaseDialect());
+            writer.println();
+            writer.printf("%d statements found, %d statements successfully parsed.", result.getNumStatementsFound(), result.getNumStatementsParsed());
+            writer.println();
+            writer.printf("%s%% of sqllogictest statements were successfully parsed.",
+                    new DecimalFormat("#.##").format(result.getNumStatementsParsed() / (double) result.getNumStatementsFound() * 100));
+        }
+        catch (FileNotFoundException e) {
+            LOG.error(e, "Could not find the result file at the given location");
+        }
+    }
+
+    private boolean isLineBeforeStatement(String line)
+    {
+        return line.startsWith("statement ok") || line.startsWith("query");
+    }
+
+    private boolean statementIsAcceptableToCurrDialect(SqlLogicTestStatement sqlLogicTestStatement)
+    {
+        if (databaseDialect == ALL) {
+            return sqlLogicTestStatement.getSkipIfSet().isEmpty() && !sqlLogicTestStatement.getCurrStatementsOnlyIfDialect().isPresent();
+        }
+        if (sqlLogicTestStatement.getCurrStatementsOnlyIfDialect().isPresent() && sqlLogicTestStatement.getCurrStatementsOnlyIfDialect().get() != databaseDialect) {
+            return false;
+        }
+        return !sqlLogicTestStatement.getSkipIfSet().contains(databaseDialect);
+    }
+
+    private boolean isConditionalRecordLine(String line)
+    {
+        String[] splitLine = line.split(" ");
+        if (splitLine.length <= 1) {
+            return false;
+        }
+        String conditionalFlag = splitLine[0];
+        if (!conditionalFlag.equals("skipif") && !conditionalFlag.equals("onlyif")) {
+            return false;
+        }
+        String potentialDialect = splitLine[1];
+        try {
+            DatabaseDialect.valueOf(potentialDialect.toUpperCase());
+        }
+        catch (IllegalArgumentException e) {
+            LOG.debug("This line had a skip-if or only-if dialect that isn't supported: %s", line);
```

Comment:
Maybe warning instead of debug? But if this is an expected occurrence, and nobody needs to care about it, then debug is fine.

Reply:
I would say it isn't an issue for the user and isn't high-priority for a maintainer of the code, so I think it's good at debug-level.